src/Loop.php

```diff
@@ -33,6 +47,20 @@ public static function execute(callable $callback, LoopDriver $driver)
     }
 
     /**
+     * Create a new driver if a factory is present, otherwise throw.
+     *
+     * @throws \LogicException if no factory is set
+     */
+    private static function createDriver()
+    {
+        if (self::$factory === null) {
+            throw new \LogicException("Can't create an event loop driver without a factory.");
+        }
+
+        return self::$factory->create();
```

Comment:
I understand that it breaks, that's why I committed a fix. I just didn't understand how it's related to _"people being able to define their own"_.